src/app/cli/src/init/client.ml

```diff
@@ -1037,36 +1069,40 @@ let pending_snark_work =
        yet"
     (Cli_lib.Background_daemon.graphql_init (return ())
        ~f:(fun graphql_endpoint () ->
-         Deferred.map
-           (Graphql_client.query_exn
-              Graphql_queries.Pending_snark_work.(make @@ makeVariables ())
-              graphql_endpoint )
-           ~f:(fun response ->
-             let lst =
-               [%to_yojson: Cli_lib.Graphql_types.Pending_snark_work.t]
-                 (Array.map
-                    ~f:(fun bundle ->
-                      Array.map bundle.workBundle ~f:(fun w ->
-                          let fee_excess_left = w.fee_excess.feeExcessLeft in
-                          { Cli_lib.Graphql_types.Pending_snark_work.Work
-                            .work_id = w.work_id
-                          ; fee_excess =
-                              Currency.Amount.Signed.of_fee
-                                (to_signed_fee_exn fee_excess_left.sign
-                                   fee_excess_left.feeMagnitude )
-                          ; supply_increase = w.supply_increase
-                          ; source_first_pass_ledger_hash =
-                              w.source_first_pass_ledger_hash
-                          ; target_first_pass_ledger_hash =
-                              w.target_first_pass_ledger_hash
-                          ; source_second_pass_ledger_hash =
-                              w.source_second_pass_ledger_hash
-                          ; target_second_pass_ledger_hash =
-                              w.target_second_pass_ledger_hash
-                          } ) )
-                    response.pendingSnarkWork )
-             in
-             print_string (Yojson.Safe.to_string lst) ) ) )
+         format_pending_snark_work graphql_endpoint ) )
+
+let pending_snark_work_range =
+  let open Command.Param in
+  let start_idx =
+    flag "--start-idx" ~aliases:[ "start-idx" ]
+      ~doc:"START_IDX first index of the range (included)"
+      (required Cli_lib.Arg_type.int16)
+  in
+  let end_idx =
+    flag "--end-idx" ~aliases:[ "end-idx" ]
+      ~doc:"END_IDX first index of the range (excluded)"
+      (optional Cli_lib.Arg_type.int16)
+  in
+  Command.async
+    ~summary:
+      "Range of snark works in JSON format that are not available in the pool \
+       yet"
+    (Cli_lib.Background_daemon.graphql_init (Args.zip2 start_idx end_idx)
+       ~f:(fun graphql_endpoint (start_idx, end_idx) ->
+         let f pending_snark_work =
+           let len_response = Array.length pending_snark_work in
+           match end_idx with
+           | None ->
+               if start_idx >= len_response then [||]
+               else Array.subo ~pos:start_idx pending_snark_work
+           | Some end_idx ->
+               let len =
+                 if len_response <= end_idx then len_response - start_idx
+                 else max 0 (end_idx - start_idx)
+               in
+               Array.sub ~pos:start_idx ~len pending_snark_work
```

Comment:
I prefer having a when clause in the match statement that verifies if start_idx <= end_idx <= len_response. And then having a catchall statement in the end that returns empty list.

Also I think this logic breaks if negative indexes are passed, which is technically possible rn.

Reply:
I pushed a a new version that checks if indexes are positives. If not, it now returns `[||]`